lib/execution_engine2/SDKMethodRunner.py

```diff
@@ -383,714 +161,248 @@ def set_log_level(self) -> Logger:
+        Get the job.
+        When as_admin, check if you have the required admin_perm or raise a Permissions Exception.
+        When requesting as a normal user, check the cache for permissions for the job.
+        If the cache contains your permission, return the job
+        If the cache doesn't contain your permission, check the job_id and workspace, and then add it to the cache
+        If you don't have permissions or are requesting a None permission, raise a Permissions Exception
+
+        :param job_id: KBase Job ID
+        :param requested_job_perm: Read or Write Access
+        :param as_admin: Check if you have admin permissions based on Permission
+        :return: The Job or Raise a Permissions Exception
+        """
+        # TODO CHeck if a valid ENUM is passed in?
+        if requested_job_perm is JobPermissions.NONE:
+            raise PermissionError(f"Requesting No Permissions for {job_id}")
+
+        if as_admin:
+            self.get_ee2_auth().check_admin_permission(
+                requested_perm=requested_job_perm
             )
-
-        if check_permission:
-            try:
-                perms = can_read_jobs(jobs, self.user_id, self.token, self.config)
-            except RuntimeError as e:
-                logging.error(
-                    f"An error occurred while checking read permissions for jobs"
-                )
-                raise e
+            job = self.get_mongo_util().get_job(job_id=job_id)
         else:
-            perms = [True] * len(jobs)
-
-        job_states = dict()
-        for idx, job in enumerate(jobs):
-            if not perms[idx]:
-                job_states[str(job.id)] = {"error": "Cannot read this job"}
-            mongo_rec = job.to_mongo().to_dict()
-            del mongo_rec["_id"]
-            mongo_rec["job_id"] = str(job.id)
-            mongo_rec["created"] = int(job.id.generation_time.timestamp() * 1000)
-            mongo_rec["updated"] = int(job.updated * 1000)
-            if job.estimating:
-                mongo_rec["estimating"] = int(job.estimating * 1000)
-            if job.running:
-                mongo_rec["running"] = int(job.running * 1000)
-            if job.finished:
-                mongo_rec["finished"] = int(job.finished * 1000)
-            if job.queued:
-                mongo_rec["queued"] = int(job.queued * 1000)
-
-            job_states[str(job.id)] = mongo_rec
-
-        job_states = OrderedDict(
-            {job_id: job_states.get(job_id, []) for job_id in job_ids}
-        )
-
-        if return_list is not None and SDKMethodRunner.parse_bool_from_string(
-            return_list
-        ):
-            job_states = {"job_states": list(job_states.values())}
-
-        return job_states
+            permission_found_in_cache = self.get_ee2_auth().get_job_permission_from_cache(
+                job_id=job_id, level=requested_job_perm
+            )
+            job = self.get_mongo_util().get_job(job_id=job_id)
```

Comment:
i guess it wasn't moved after `test_job_permission` becuase `test_job_permission` also takes a `job`?